geomesa-tutorials-common/src/main/java/org/geomesa/example/quickstart/GeoMesaQuickStart.java

```diff
@@ -125,40 +107,63 @@ public DataStore createDataStore(Map<String, String> params) throws IOException
         System.out.println();
-        return sft;
+    }
+
+    public List<SimpleFeature> getTestFeatures(TutorialData data) {
+        System.out.println("Generating test data");
+        List<SimpleFeature> features = data.getTestData();
+        System.out.println();
+        return features;
+    }
+
+    public List<Query> getTestQueries(TutorialData data) {
+        return data.getTestQueries();
     }
 
     public void writeFeatures(DataStore datastore, SimpleFeatureType sft, List<SimpleFeature> features) throws IOException {
-        System.out.println("Writing test data");
-        // use try-with-resources to ensure the writer is closed
-        try (FeatureWriter<SimpleFeatureType, SimpleFeature> writer =
-                 datastore.getFeatureWriterAppend(sft.getTypeName(), Transaction.AUTO_COMMIT)) {
-            for (SimpleFeature feature : features) {
-                // using a geotools writer, you have to get a feature, modify it, then commit it
-                // appending writers will always return 'false' for haveNext, so we don't need to bother checking
-                SimpleFeature toWrite = writer.next();
-                // copy attributes
-                toWrite.setAttributes(feature.getAttributes());
-                // updating the feature ID requires casting to an implementation class
-                // alternatively, you can use the PROVIDED_FID hint in the user data
-                ((FeatureIdImpl) toWrite.getIdentifier()).setID(feature.getID());
-                // make sure to copy the user data, if there is any
-                toWrite.getUserData().putAll(feature.getUserData());
-                // write the feature
-                writer.write();
+        if (features.size() > 0) {
+            System.out.println("Writing test data");
+            // use try-with-resources to ensure the writer is closed
+            try (FeatureWriter<SimpleFeatureType, SimpleFeature> writer =
+                     datastore.getFeatureWriterAppend(sft.getTypeName(), Transaction.AUTO_COMMIT)) {
+                for (SimpleFeature feature : features) {
+                    // using a geotools writer, you have to get a feature, modify it, then commit it
+                    // appending writers will always return 'false' for haveNext, so we don't need to bother checking
+                    SimpleFeature toWrite = writer.next();
+                    // copy attributes
+                    toWrite.setAttributes(feature.getAttributes());
+                    // updating the feature ID requires casting to an implementation class
+                    // alternatively, you can use the PROVIDED_FID hint in the user data
+                    ((FeatureIdImpl) toWrite.getIdentifier()).setID(feature.getID());
+                    // make sure to copy the user data, if there is any
+                    toWrite.getUserData().putAll(feature.getUserData());
+                    // write the feature
+                    writer.write();
```

Comment:
Do we close writers?

Reply:
on line 143-144 it uses try-with-resources to ensure it gets closed, and has a comment to emphasize it

Reply:
Oh, right, my bad